certbot/certbot/_internal/client.py

```diff
@@ -548,8 +548,10 @@ def _retry_obtain_certificate(self, key: util.Key,
         display_util.notify("Unable to obtain a certificate with every requested "
             f"domain. Retrying without: {domains_list}")
         if not self.config.dry_run:
-            os.remove(key.file)
-            os.remove(csr.file)
+            if key.file:
+                os.remove(key.file)
+            if csr.file:
+                os.remove(csr.file)
```

Comment:
TODO: pretty sure can delete this code, since we are not saving either of these anymore.